NSrtm.Core/Pgm/DataDesciption/PgmDataDescriptionExtractor.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace NSrtm.Core.Pgm
+{
+    internal static class PgmDataDescriptionExtractor
+    {
+        private static readonly Regex _preambleParser =
+            new Regex(
+                @"^(P5)[\S\s]+(?=Offset)[^0-9-]+(?<offset>-?[0-9]+)[\S\s]+(?=Scale)[^0-9-]+(?<scale>-?[0-9]+\.?[0-9]+)[\S\s]+(?=Origin)[^0-9]+(?<lat>[0-9]+)[^0-9]+(?<lon>[0-9]+)[^0-9]+\S+\s(?<width>\d+)\D+(?<height>\d+)\D+(?<max>\d+)$");
+
+        public static PgmDataDescription FromStream(Stream stream)
+        {
+            var preamble = extractPreambleFromStream(stream);
+            return getConstatantsFromPreamble(preamble);
+        }
+
+        private static string extractPreambleFromStream(Stream stream)
+        {
+            using (var reader = new StreamReader(stream, Encoding.UTF8, false, 1024, true))
+            {
+                var preamble = new StringBuilder(reader.ReadLine());
+                string lastLine;
+                do
+                {
+                    lastLine = reader.ReadLine();
+                    preamble.Append(lastLine + "\n");
```

Comment:
Why already read line is not processed to find if it contains sth interesting (only buffered)? Again you callect whole preamble to variable and process later... why?